sql/src/main/java/io/crate/expression/scalar/postgres/CurrentSettingFunction.java

```diff
@@ -0,0 +1,144 @@
+                                              .builder(Param.STRING)
+                                              .withVarArgs(Param.BOOLEAN)
+                                              .limitVarArgOccurrences(1)
+                                              .build()) {
+                @Override
+                public FunctionImplementation getForTypes(List<DataType> datatypes) {
+                    return new CurrentSettingFunction(
+                        new FunctionInfo(new FunctionIdent(FQN, datatypes), datatypes.get(0))
+                    );
+                }
+            });
+    }
+
+    private final FunctionInfo info;
+
+    CurrentSettingFunction(FunctionInfo info) {
+        this.info = info;
+    }
+
+    @Override
+    public FunctionInfo info() {
+        return info;
+    }
+
+    @Override
+    public Symbol normalizeSymbol(Function function, @Nullable TransactionContext txnCtx) {
+        if (txnCtx == null) {
+            return function;
+        }
+
+        if (anyNonLiterals(function.arguments())) {
+            return function;
+        }
+
+        final int args = function.arguments().size();
+        assert args == 1 || args == 2 : "function's number of arguments must be 1 or 2";
+
+        final Symbol settingNameSymbol = function.arguments().get(0);
+        final String settingName = (String) ((Input) settingNameSymbol).value();
+        if (settingName == null) {
+            return Literal.NULL;
+        }
+
+        Boolean missingOk = Boolean.FALSE;
+        if (args == 2) {
+            final Symbol missingOkSymbol = function.arguments().get(1);
+            missingOk = (Boolean) ((Input) missingOkSymbol).value();
+            if (missingOk == null) {
+                return Literal.NULL;
+            }
+        }
+        return Literal.of(eval(txnCtx, settingName, missingOk));
+    }
+
+    @Override
+    public String evaluate(TransactionContext txnCtx, Input<Object>... args) {
+        assert args.length == 1 || args.length == 2 : "number of args must be 1 or 2";
+
+        final String settingName = (String) args[0].value();
+        if (settingName == null) {
+            return null;
+        }
+
+        final Boolean missingOk = (args.length == 2)
+            ? (Boolean) args[1].value()
+            : Boolean.FALSE;
+        if (missingOk == null) {
+            return null;
+        }
+        return eval(txnCtx, settingName, missingOk);
+    }
+
+    private String eval(@Nonnull TransactionContext txnCtx,
+                        @Nonnull String settingName,
+                        boolean missingOk) {
+        final SessionSetting<?> sessionSetting = SessionSettingRegistry.SETTINGS.get(settingName);
+        if (sessionSetting == null) {
+            if (missingOk) {
+                return null;
+            } else {
+                throw new IllegalArgumentException(
+                    String.format(Locale.ENGLISH, "Unrecognised Setting: %s", settingName));
```

Comment:
```suggestion
                    "Unrecognised Setting: " + settingName);
```

would be cheaper than the `String.format` variant and as readable.